querygen/querygen.go

```diff
@@ -140,3 +140,22 @@ func GenReleaseCompletedQuery(index string) string {
 	)
 	return q
 }
+
+// GenPullRequestCreation returns a Splunk query for generating Segment events
+// whenever a PR is created in the users GitHub repository.
+func GenPullRequestCreation(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="appstudio.redhat.com" `+
+			`"objectRef.resource"="components" `+
+			`"responseObject.metadata.annotations.build.appstudio.openshift.io/status"="*pac*" `+
+			`"responseObject.metadata.generation"="1"  `+
```

Comment:
Have you tested scenarios where this generation is incremented prior to PR creation? For example, the user might create a component using the default build pipeline but then upgrade to a custom pipeline definition. I'm concerned this approach is too brittle when we can't predict the order of the end user's actions.

Reply:
@amisstea Comparing the components before and after the PR was merged, there were only a few differences, namely the generation got updated from 1 to 2 and the status logs got updated. So I am filtering out `"responseObject.metadata.generation"="1"` for PR creation. Will check if I can something more meaningful